### DISCUSSION TITLE
Using external certificate?

### DISCUSSION THREAD

**Discussion by parasiteoflife** [NginxProxyManager/nginx-proxy-manager#3999] [Q&A]
Is there a way to use an external certificate?
I'm using mailcow which generates its own certs so I need to use their certificates in NPM, how can I do this? Even if I somehow create a new cert in NPM, then replace the files with mailcow's certs, NPM will try to renew the certificates which will break things.

**Comment by shalafi99** [comment]
For a certificate generated/managed outside of NPM, I understand you can use "Custom" under the "Add SSL Certificate" action/button:

![image](https://github.com/user-attachments/assets/26d74fc2-5893-4548-a269-827335befcc6)

It expects you to upload the certificate files from the web interface itself, instead of referencing their path within the filesystems available to NPM

**Comment by madmalkav** [comment]
@parasiteoflife what did you end up doing? I'm also in a situation where I'm going to want to share certs, either from the system to the NPM docker, or get NPM to manage the certs for the domain but share systemwide, and I don't know how to do either 😅
